### PR TITLE
docs: write_prediction methods documentation

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/5650
Adds missing documentation for lightningmodules `write_prediction` and `write_prediction_dict` methods

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — COMMENTED** [review]
well in fact it is adding two new `write_prediction` and `write_prediction_dict` methods

**Comment by SkafteNicki** [comment]
@Borda the methods were there already, it was just the documentation missing :]

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
> @Borda the methods were there already, it was just the documentation missing :]

yeah, I have overlooked it, sorry

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -348,10 +348,46 @@ def log_dict(
...
+        Write a dictonary of predictions to disk at once using ``torch.save``
+
+        Example::
+
+            pred_dict = {'pred1': torch.tensor(...), 'pred2': torch.tensor(...)}
+            self.write_prediction_dict(pred_dict)
+
+        Args:
+            predictions_dict: dict containing predictions, where predicions each prediction should
+                either be single :class:`~torch.Tensor` or a list of them
```
Comment (awaelchli) [inline pytorch_lightning/core/lightning.py:384]:
```suggestion
            predictions_dict: dict containing predictions, where each prediction should
                either be single :class:`~torch.Tensor` or a list of them
```
not sure if typo

**Review by awaelchli — APPROVED** [review]
the long lost docs have returned
